Topic 1/src/exercise1to6/cart/Transaction.java

```diff
@@ -0,0 +1,45 @@
+package exercise1to6.cart;
+
+import exercise1to6.payments.Payment;
+
+public class Transaction {
+	private static int idTransaction = 1;
+	private static int ID;
+	private Payment payment;
+	private double total;
+
+	public Transaction(double total, Payment payment) {
+		this.setPayment(payment);
+		this.setTotal(total);
+		this.ID = this.getNewIDTransaction();
+	}
+
+	private int getNewIDTransaction() {
+		return idTransaction++;

```

Comment:
You need to create a thread sage incremental id generator in order to provide unique Id to Payment transactions.